pkg/util/strategicpatch/patch.go

```diff
@@ -31,7 +31,352 @@ import (
 // lists should be merged or replaced.
 //
 // For more information, see the PATCH section of docs/api-conventions.md.
+
+// CreateStrategicMergePatch creates a patch that can be passed to StrategicMergePatch.
```

Comment:
So, based on all three of our examples, are we concluding that we don't need any additional functionality beyond what's already implemented? In other words, there's no need for any kind of conflict detection between the patch and the server - the generated patch should always just clobber what's on the server because there's no way to tell what would be a deliberate overriding of a default-filled-in value (which should not be a special case or require a force), versus a deliberate overriding of an automated-process-modified value (which arguably shouldn't be treated any differently than specifying a previously-defaulted field).

It seems to be that always-clobbering is the right approach here. If that would result in an invalid object state, server-side validation can throw an error and the user can fix their local config (or send a delete to the server) to get to a good state.